.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,201 @@
+          AZURE_ENABLE_GLOBAL_DOCUMENT_ACCESS: ${{ vars.AZURE_ENABLE_GLOBAL_DOCUMENT_ACCESS }}
+          AZURE_ENABLE_UNAUTHENTICATED_ACCESS: ${{ vars.AZURE_ENABLE_UNAUTHENTICATED_ACCESS }}
+          AZURE_AUTH_TENANT_ID: ${{ vars.AZURE_AUTH_TENANT_ID }}
+          AZURE_SERVER_APP_ID: ${{ vars.AZURE_SERVER_APP_ID }}
+          AZURE_CLIENT_APP_ID: ${{ vars.AZURE_CLIENT_APP_ID }}
+          ALLOWED_ORIGIN: ${{ vars.ALLOWED_ORIGIN }}
+          AZURE_ADLS_GEN2_STORAGE_ACCOUNT: ${{ vars.AZURE_ADLS_GEN2_STORAGE_ACCOUNT }}
+          AZURE_ADLS_GEN2_FILESYSTEM_PATH: ${{ vars.AZURE_ADLS_GEN2_FILESYSTEM_PATH }}
+          AZURE_ADLS_GEN2_FILESYSTEM: ${{ vars.AZURE_ADLS_GEN2_FILESYSTEM }}
+
+        steps:
+          #--------------------------Setup--------------------------------
+
+          - name: Checkout
+            uses: actions/checkout@v4
+
+          - name: Setup python
+            uses: actions/setup-python@v5
+            with:
+              python-version: ${{ matrix.python_version }}
+              architecture: x64
+    
+          - name: Install azd
+            uses: Azure/setup-azd@v1.0.0
+    
+          - name: Install Nodejs
+            uses: actions/setup-node@v4
+            with:
+              node-version: 18
+
+          #--------------------------Deploy--------------------------------
+          - name: Check credentials
+            run: |
+              echo "AZURE_CLIENT_ID: ${{ env.AZURE_CLIENT_ID }}"
+              echo "AZURE_TENANT_ID: ${{ env.AZURE_TENANT_ID }}"
+              echo "AZURE_SUBSCRIPTION_ID: ${{ env.AZURE_SUBSCRIPTION_ID }}"
+              echo "AZURE_PRINCIPAL_ID: ${{ secrets.AZURE_PRINCIPAL_ID }}"
+
+          - name: Azure login
+            uses: azure/login@v2
+            with:
+              client-id: ${{ env.AZURE_CLIENT_ID }}
+              tenant-id: ${{ env.AZURE_TENANT_ID }}
+              subscription-id: ${{ env.AZURE_SUBSCRIPTION_ID }}
+            
+          - name: Azure CLI script
+            uses: azure/cli@v2
+            with:
+              azcliversion: latest
+              inlineScript: |
+                az account show
+
+          #--------------------------Run Evaluation--------------------------------
+          - name: Install dependencies
+            working-directory: ./evaluation
+            run: |
+              python -m pip install --upgrade pip
+              pip install -r scripts/requirements.txt
+              pip install pyrit
+          - name: Run AI Rag Evaluations
+            working-directory: ./evaluation
+            run: |
+              OPENAI_HOST=$OPENAI_HOST OPENAI_GPT_MODEL=$OPENAI_GPT_MODEL AZURE_OPENAI_EVAL_DEPLOYMENT=$AZURE_OPENAI_EVAL_DEPLOYMENT AZURE_OPENAI_SERVICE=$AZURE_OPENAI_SERVICE AZURE_SEARCH_SERVICE=$AZURE_SEARCH_SERVICE AZURE_SEARCH_INDEX=$AZURE_SEARCH_INDEX AZURE_PRINCIPAL_ID=$AZURE_PRINCIPAL_ID python -m scripts evaluate --config=./config.json --numquestions=10
```

Comment:
Perhaps there's a more elegant way of setting shell/environment variables!

Reply:
Yes there should be, I'll look it up.